sycl/include/sycl/ext/intel/fpga_device_selector.hpp

```diff
@@ -46,25 +63,44 @@ static constexpr auto EMULATION_PLATFORM_NAME =
 static constexpr auto HARDWARE_PLATFORM_NAME =
     "Intel(R) FPGA SDK for OpenCL(TM)";
 
-class fpga_selector : public platform_selector {
+int fpga_selector_v(const device &device) {
+  return detail::selectDeviceByPlatform(HARDWARE_PLATFORM_NAME, device);
+}
+
+int fpga_emulator_selector_v(const device &device) {
+  return detail::selectDeviceByPlatform(EMULATION_PLATFORM_NAME, device);
+}
+
+int fpga_simulator_selector_v(const device &device) {
+  static bool IsFirstCall = true;
+  if (IsFirstCall) {
+    detail::enableFPGASimulator();
+    IsFirstCall = false;
+  }
```

Comment:
Not thread safe, I think. We might have two threads accessing this and the second might reach line 80 before the first one finishes initialization.

Reply:
You are right, but I do not believe it is worth the overhead of making it thread-safe. Worst case we have two threads setting the environment variable to 1, which is a wasted syscall, but not breaking anything.

Reply:
Yes, you're right - I misread it at first.